eth/api.go

```diff
@@ -636,3 +636,33 @@ func storageRangeAt(st state.Trie, start []byte, maxResult int) StorageRangeResu
 	}
 	return result
 }
+
+func (api *PrivateDebugAPI) GetDirtyAccountsByNumber(startNum, endNum uint64) ([]common.Address, error) {
+	return api.getDirtyAccounts(api.eth.blockchain.GetBlockByNumber(startNum), api.eth.blockchain.GetBlockByNumber(endNum))
+}
+
+func (api *PrivateDebugAPI) GetDirtyAccountsByHash(startHash, endHash common.Hash) ([]common.Address, error) {
+	return api.getDirtyAccounts(
+		api.eth.blockchain.GetBlockByHash(startHash),
+		api.eth.blockchain.GetBlockByHash(endHash))
+}
+
+func (api *PrivateDebugAPI) getDirtyAccounts(startBlock, endBlock *types.Block) ([]common.Address, error) {
+	oldTrie, err := trie.NewSecure(startBlock.Root(), api.eth.chainDb, 0)
```

Comment:
`startBlock` and `endBlock` might not exist here if the user specifies a number > head, or a non existent hash. That will cause `startBlock.Root()` and `endBlock.Root()` to panic.